presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/StaticSelector.java

```diff
@@ -43,31 +55,86 @@ public StaticSelector(Optional<Pattern> userRegex, Optional<Pattern> sourceRegex
+    @Override
+    public Optional<SelectionContext<VariableMap>> match(SelectionCriteria criteria)
     {
-        if (userRegex.isPresent() && !userRegex.get().matcher(context.getUser()).matches()) {
-            return Optional.empty();
+        Map<String, String> variables = new HashMap<>();
+
+        if (userRegex.isPresent()) {
+            Matcher userMatcher = userRegex.get().matcher(criteria.getUser());
+            if (!userMatcher.matches()) {
+                return Optional.empty();
+            }
+
+            addVariableValues(userRegex.get(), criteria.getUser(), variables);
         }
+
         if (sourceRegex.isPresent()) {
-            String source = context.getSource().orElse("");
+            String source = criteria.getSource().orElse("");
             if (!sourceRegex.get().matcher(source).matches()) {
                 return Optional.empty();
             }
+
+            addVariableValues(sourceRegex.get(), source, variables);
         }
-        if (!clientTags.isEmpty() && !context.getTags().containsAll(clientTags)) {
+
+        if (!clientTags.isEmpty() && !criteria.getTags().containsAll(clientTags)) {
             return Optional.empty();
         }
 
         if (queryType.isPresent()) {
-            String contextQueryType = context.getQueryType().orElse("");
+            String contextQueryType = criteria.getQueryType().orElse("");
             if (!queryType.get().equalsIgnoreCase(contextQueryType)) {
                 return Optional.empty();
             }
         }
 
-        return Optional.of(group.expandTemplate(context));
+        variables.putIfAbsent(USER_VARIABLE, criteria.getUser());
+
+        // Special handling for source, which is an optional field that is part of the standard variables
+        variables.putIfAbsent(SOURCE_VARIABLE, criteria.getSource().orElse(""));
+
+        VariableMap map = new VariableMap(variables);
+        ResourceGroupId id = group.expandTemplate(map);
+
+        return Optional.of(new SelectionContext<>(id, map));
+    }
+
+    private static void addNamedGroups(Pattern pattern, HashSet<String> variables)
+    {
+        Matcher matcher = NAMED_GROUPS_PATTERN.matcher(pattern.toString());
+        while (matcher.find()) {
+            String name = matcher.group(1);
+            checkArgument(!variables.contains(name), "Multiple definitions found for variable ${" + name + "}");
+            variables.add(name);
+        }
+    }
+
+    private void addVariableValues(Pattern pattern, String candidate, Map<String, String> mapping)
+    {
+        for (String key : variableNames) {
+            Matcher keyMatcher = pattern.matcher(candidate);
+            if (keyMatcher.find()) {
+                try {
+                    String value = keyMatcher.group(key);
+                    if (value != null) {
+                        mapping.put(key, value);
+                    }
+                }
+                catch (IllegalArgumentException ignored) {
+                }
```

Comment:
Add comment to explain when this happens. It's a good idea to avoid empty `catch` clause without comments.